### ISSUE TITLE
Feature: Paint gun hud

### ISSUE THREAD

**Issue by JJL772** [ChaosInitiative/Portal-2-Community-Edition#705]
### Which component should be improved?

UI/UX

### Describe your suggestion

The current paint gun hud is written in vgui and is barely functional. This should be replaced with a proper paint gun hud element in panorama. 

### Expected result

A paint gun hud element

**Comment by JJL772** [comment]
A couple ideas:
![image](https://user-images.githubusercontent.com/19717056/130712202-813d9232-befe-4add-a538-d2e0989d7326.png)

![image](https://user-images.githubusercontent.com/19717056/130712267-c4839c1f-17f0-4e4b-bdd2-2aea322dad9b.png)


For the crosshair:
![image](https://user-images.githubusercontent.com/19717056/130712232-db2c787e-b010-4874-892a-5e4976debfec.png)

Crosshair with limited paint:
![image](https://user-images.githubusercontent.com/19717056/130712255-c55ca95e-f058-4b66-bae8-2524efecfe99.png)

**Comment by Luke18033** [comment]
The second HUD there is kind of interesting, though I feel like it doesn't really fit Portal 2/P2CE's UI style, and not having all the gels visible at once would be a little annoying. So the first HUD would probably be better overall. As for the crosshair, we should probably do a combination of those two, with multiple bars, each of which fills up depending on how much of that gel you have left.

**Comment by Luke18033** [comment]
Also why is the adhesion gel green...?

**Comment by JJL772** [comment]
The first hud is my favorite. We could also make it "slide down" to mostly conceal itself after a gel has been selected. Pressing the nextpaint/prevpaint key would cause those hud elements to "slide up" revealing more details.

I hope I'm explaining this thoroughly :grin:

**Comment by JJL772** [comment]
Also, for the crosshair, it might be best to have a hl2-style bar that drains as the paint is used up or something. The bar's color can correspond to the paint type, although that might be an accessibility issue for colorblind people.

**Comment by Luke18033** [comment]
Yeah, that's a good point. Maybe it would be better to only show the selected gel then, closer to the second crosshair.

**Comment by Luke18033** [comment]
You'd be able to see how much of the other gels you had remaining through the full HUD in the corner (which has icons for the gel types).

**Comment by JJL772** [comment]
I've created a discussion #706 to discuss this :)
